### ISSUE TITLE
Illegal Offset w:compat 

### ISSUE THREAD

**Issue by shad0wfir3** [PHPOffice/PHPWord#1024]
Issue found when using latest build on PHP7. Reference to #927

Found a empty string in: 
PHPOffice\PHPWord\Writer\Word2007\Part\Settings     

**getSettings():109**

```
private function getSettings()
    {
           .......

            'w:compat' => ' ',

           ........
    }
```

Updated to

```
private function getSettings()
    {
           .......

            'w:compat' => array('@attributes' => array('w:val' => '')),

           ........
    }
```

And it seems to be working. Not sure if this is supposed to be used in the same way though, but this has allowed me to export and generate word docs.

Can anyone who knows more check and see if this is in fact a issue they are experiencing as well, otherwise, is this fix a valid fix for the problem?

**Comment by karinegomes** [comment]
This is an issue with PHP 7, getting it here too. You can just replace '' for array() and it will also work fine.
Not sure why they haven't merged it yet.

**Comment by SailorMax** [comment]
More looks like issue with PHPWord.
In function PhpOffice\PhpWord\Writer\Word2007\Part\Settings\getCompatibility() we can see:
``` php
$this->settings['w:compat']['w:compatSetting'] = array(...)
```
How? if `$this->settings['w:compat']` is string?

Same problem in PhpOffice\PhpWord\Writer\Word2007\Part\Settings\writeSetting():
``` php
foreach ($settingValue as $childKey => $childValue) { ... }
```
When `$settingValue` in case with 'w:compat' is empty **string**...

PHP 7.x just more strict than early.

**Comment by ezintz** [comment]
https://github.com/PHPOffice/PHPWord/issues/927 - Please release a fix, thank you in advance!

**Comment by mikebronner** [comment]
Waiting on this as well. Honestly quite surprised that this project hasn't been updated in over a year -- it seems like a great endeavor, clearly filling a niche. Like @ezintz, stale projects don't provide a good confidence level to roll them out into production. Hope you guys are able to update soon, looking forward to it. :)

**Comment by troosan** [comment]
fixed in 0.14
